### PR TITLE
Mon 147075 otel engine passive checks

### PR DESCRIPTION
**Pull request by jean-christophe81**
REFS: MON-147075
## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [X] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/resources/Engine.py [hunk 1]
```diff
@@ -2465,6 +2465,28 @@ def ctn_set_services_passive(poller: int, srv_regex):
...
+def ctn_set_hosts_passive(poller: int, host_regex):
+    """
+    Set passive a list of hosts.
+
+    Args:
+        poller (int): Index of the poller to work with.
+        srv_regex (str): A regexp to match host name.
+    """
+
+    with open("{}/config{}/hosts.cfg".format(CONF_DIR, poller), "r") as ff:
```
Comment (bouda1) [inline tests/resources/Engine.py:2477]:
```suggestion
    with open(f"{CONF_DIR}/config{poller}/hosts.cfg", "r") as ff:
```

File: engine/tests/opentelemetry/open_telemetry_test.cc [hunk 2]
```diff
@@ -57,35 +60,34 @@ extern const char* telegraf_example;
...
+//   static std::shared_ptr<open_telemetry> load(
+//       const std::string_view& config_path,
+//       const std::shared_ptr<asio::io_context>& io_context,
+//       const std::shared_ptr<spdlog::logger>& logger) {
+//     std::shared_ptr<open_telemetry> ret =
+//         std::make_shared<open_telemetry>(config_path, io_context, logger);
+//     ret->_reload();
+//     return ret;
+//   }
+// };
```
Comment (bouda1) [inline engine/tests/opentelemetry/open_telemetry_test.cc:90]: to remove ?

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/otl_check_result_builder.hh [hunk 3]
```diff
@@ -19,35 +19,40 @@
...
+  bool operator()(const otl_data_point& left, uint64_t right) const {
+    return left.get_nano_timestamp() < right;
+  }
+
+  bool operator()(uint64_t left, const otl_data_point& right) const {
+    return left < right.get_nano_timestamp();
+  }
 };
 
+class name_to_metrics
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/otl_check_result_builder.hh:50]: done

File: CMakeListsWindows.txt [hunk 4]
```diff
@@ -16,12 +16,15 @@
...
+#in order to make fmt compile
+add_definitions("/utf-8")
+
 # When we build from cache (CI), we don't use vcpkg cmaketool, so we tell to cmake where to find packages info
 if (BUILD_FROM_CACHE)
   LIST(APPEND CMAKE_PREFIX_PATH "build_windows/vcpkg_installed/x64-windows")
 endif()
 
-#in order to make fmt compile version >= 2024.07.12
+#in order to make fmt compile
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply CMakeListsWindows.txt:27]: done

File: engine/inc/com/centreon/engine/checks/checker.hh [hunk 5]
```diff
@@ -92,8 +94,19 @@ class checker : public commands::command_listener {
...
+ * @brief allow to inspect _to_reap_partial
+ *
+ * @tparam queue_handler
+ * @param handler must have () (const std::deque<check_result::pointer>  &)
+ */
+template <class queue_handler>
+void checker::inspect_reap_partial(queue_handler&& handler) const {
+  std::lock_guard<std::mutex> lock(_mut_reap);
+  handler(_to_reap_partial);
 }
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/inc/com/centreon/engine/checks/checker.hh:108]: done

File: tests/resources/Engine.py (hunk 1 above)
Comment (jean-christophe81) [reply tests/resources/Engine.py:2477]: done

File: engine/tests/opentelemetry/open_telemetry_test.cc (hunk 2 above)
Comment (jean-christophe81) [reply engine/tests/opentelemetry/open_telemetry_test.cc:90]: done

**Review by bouda1 — APPROVED** [review]

**Pull request merged**